src/fidesops/service/connectors/saas_connector.py

```diff
@@ -76,15 +74,20 @@ def query_config(self, node: TraversalNode) -> SaaSQueryConfig:
 
     def test_connection(self) -> Optional[ConnectionTestStatus]:
         """Generates and executes a test connection based on the SaaS config"""
+
+        test_request_path = self.saas_config.test_request.path
+        prepared_request: SaaSRequestParams = (test_request_path, {}, {})
+
         try:
-            test_request_path = self.saas_config.test_request.path
-            prepared_request: SaaSRequestParams = (test_request_path, {}, {})
-            self.client().get(prepared_request)
+            response = self.client().get(prepared_request)
         except Exception:
             raise ConnectionException(
-                f"Operational Error connecting to {self.configuration.key}."
+                f"Operational Error connecting to '{self.configuration.key}'."
             )
 
+        if not response.ok:
+            raise ClientUnsuccessfulException(status_code=response.status_code)
+
```

Comment:
Nice catch, I looked this up in the Mailchimp API documentation and apparently we can authenticate using one of two ways.

1. Bearer auth using the api_key
2. Basic auth with "anystring" as the username, and the api_key as the password

I'll keep the basic auth for now but I think this would be cleaner if we used bearer auth in the future to prevent confusion.

Reply:
cool thanks for digging into that